src/navigator-card-body.jsx

```diff
@@ -412,10 +349,95 @@ export const NavigatorCardBody = ({
                   className="pf-m-no-border-rows"
                   variant="compact"
                   columns={[_("Name")]}
-                  rows={sortedFiles.map(file => ({ columns: [{ title: <Item file={file} key={file.name} /> }] }))}
+                  rows={sortedFiles.map(file => ({
+                      columns: [
+                          {
+                              title:
+    <Item
+      file={file} key={file.name}
+      isSelected={!!selected.find(s => s.name === file.name)}
+      setSelected={setSelected}
+      onDoubleClickNavigate={onDoubleClickNavigate}
+      isGrid={isGrid}
+    />
+                          }
+                      ]
+                  }))}
                   onContextMenu={handleContextMenu}
                 />
             </>
         );
     }
 };
+
+const Item = React.memo(function Item({ file, isSelected, setSelected, onDoubleClickNavigate, isGrid }) {
+    function handleClick(ev, file) {
+        if (ev.detail > 1) {
+            onDoubleClickNavigate(file);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-303-20240306-140502-d545e596-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)